### PR TITLE
Storing the definition's model file name in RefRulePosition

### PR DESCRIPTION
**Pull request by hlouzada**
## Details
Feature implementation for Issue #277

## Code review checklist

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [x] Tests have been included and/or updated
- [x] Docstrings have been included and/or updated, as appropriate
- [x] Standalone docs have been updated accordingly
- [x] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`, no need
      to update for typo fixes and such).

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by goto40** [comment]
Thank you for your contribution. 

I assume you need this PR for model editing purposes?

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ please take a look at related PRs and issues and see if the change affects you.
 
 ### Changed
 
+- Added `def_file_name` atribute to `RefRulePosition` for storing the definition's
+  model file name in case of cross-references between models. ([#277])
```
Comment (goto40) [inline CHANGELOG.md]: Minor: you need to add the reference at the end of the file.

**Review by goto40 — COMMENTED** [review]
Technically, your changes look OK, after a first quick check... Do you think you can sketch the use case related to your change. It would ease the review.

File: CHANGELOG.md (hunk 1 above)
Comment (hlouzada) [reply CHANGELOG.md]: Should I also reference the pull request?

Comment (goto40) [reply CHANGELOG.md]: Yes. This would be great.

**Comment by goto40** [comment]
To me the PR looks good, so far. @igordejanovic any comments?

@hlouzada : I am looking forward to see what you put together... Thank you for your interest and your work+time!

**Review by goto40 — APPROVED** [review]

**Comment by hlouzada** [comment]
> Technically, your changes look OK, after a first quick check... Do you think you can sketch the use case related to your change. It would ease the review.

A use case would be using an IDE's go-to definition that uses the model's attribute _pos_crossref_list, enabled by textx-tools, on a language that its grammar allows cross models imports. Thus, the goto-definition needs to know the position of the definition and which model it's from.

Take this grammar as an example:

```re
Model:
        imports*=Import
        packages*=Package
;
Package:
        'package' name=ID '{'
            objects*=Object
        '}'
;
Object:
    'object' name=ID ('ref' ref=[Object|FQN])?
;
FQN: ID+['.'];
FQNI: ID+['.']('.*')?;
Import: 'import' importURI=FQNI;
```

Now a model named ``./folder/A.model``:
```
package packageA1 {
    object A
}
package packageA2 {
    object A ref packageA1.A
    object B ref A
}
```
And another model named ``./B.model``:
```
import folder.A

package packageB {
    object A1 ref packageA1.A
    object A2 ref packageA2.A
}
```
I would like to know, not only the position in the model but also, the model's path of the ``packageA1.A`` and ``packageA2.A`` definition.

**Comment by igordejanovic** [comment]
@hlouzada Looks good! Thanks for the contribution.

**Pull request merged**